swagger-api-client/publish.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -e
+
+# sbt publish publishes libs to Artifactory for the scala version sbt is running as.
+# sbt +publish publishes libs to Artifactory for all scala versions listed in crossScalaVersions.
+SNAPSHOT=$(( "$TRAVIS_PULL_REQUEST" == "false" && "$TRAVIS_BRANCH" == "develop" ? "SNAP" : "" ))
+
+echo "getting swagger codegen jar..."
+wget http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O swagger-codegen-cli.jar
+echo "generating Sam Java Client API..."
+java -jar swagger-codegen-cli.jar generate -l java --input-spec ./src/main/resources/swagger/api-docs.yaml --output generated
+
+COMMIT_HASH="${TRAVIS_COMMIT:0:7}"
+VERSION_HASH="1.0-${COMMIT_HASH}${SNAPSHOT}"
+ORGANIZATION="org.broadinstitute.dsde"
+APP_NAME="sam"
+PATH_PREFIX="org/broadinstitute/dsde/$APP_NAME"
+SCALA_11_VERSION="2.11.8"
+SCALA_11_VERSION_SHORT="${SCALA_11_VERSION:0:4}"
+SCALA_12_VERSION="2.12.3"
+SCALA_12_VERSION_SHORT="${SCALA_12_VERSION:0:4}"
+SCALA_VERSIONS="List(\"$SCALA_11_VERSION\", \"$SCALA_12_VERSION\")"
```

Comment:
Blerg. Can you decide what to publish by finding the poms / jars etc and deconstructing the path instead? Defining your list of scala version here and poking it in `crossScalaVersions` feels a bit gross.